judge/views/contests.py

```diff
@@ -122,6 +122,8 @@ def get_context_data(self, **kwargs):
         context['current_contests'] = present
         context['future_contests'] = future
         context['finished_contests'] = finished
+        context['LIVE'] = ContestParticipation.LIVE
+        context['SPECTATE'] = ContestParticipation.SPECTATE
```

Comment:
Can we make this `context['is_spectator'] = ...` instead? Passing constants like this into the template feels a little icky.

Reply:
Nevermind, we can't do this because uh, there are multiple participations...

So we should make a method on the object.

Reply:
There's a world in which we refactor this out to an enum to make this all a little nicer, but the fix implemented here is just way easier.